### DISCUSSION TITLE
🌟 Request for feedback: If you haven't tried Copilot yet, why not?

### DISCUSSION THREAD

**Discussion by jayceeday** [community/community#64222] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hello GitHub community! I’m Jaycee and I’m working at GitHub on Copilot.

A question for people who **haven’t** used Copilot yet: Why **haven’t** you tried Copilot? 

Please let me know your honest answers by replying to this discussion 🙏

  Thanks!

**Comment by Jaber-Saed** [comment]
Hi Jaycee,

It's great to hear that you're working on GitHub Copilot! While I can't provide real-time community feedback, I can offer some potential reasons why some people might not have tried Copilot yet:

1. **Awareness**: Some users might not be aware of Copilot's existence or its capabilities.

2. **Learning Curve**: There could be concerns about the learning curve associated with adopting a new tool or plugin.

3. **Workflow Compatibility**: Users might have established coding workflows that don't currently include a tool like Copilot.

4. **Cost Considerations**: Depending on the pricing structure, users might have cost considerations, especially for larger teams or personal projects.

5. **Project Specificity**: Some users might work on projects that don't align with the current capabilities of Copilot.

6. **Privacy and Security**: Concerns about sharing proprietary code or exposing sensitive information could be a factor.

7. **Editor Preferences**: Users might have strong preferences for certain text editors or integrated development environments.

8. **Customization Needs**: Some developers might require a high degree of customization in their code generation, which Copilot might not fully support.

9. **Testing and Trust**: Users might want to thoroughly test Copilot before integrating it into their projects and workflows.

10. **Manual Coding Desire**: Some developers may prefer manually writing code to better understand the logic and maintain full control.

11. **Existing Codebase**: For projects with extensive existing codebases, transitioning to Copilot might be a gradual process.

It's important to consider these factors and address them when promoting Copilot to potential users. Providing clear information about the benefits, addressing concerns, and offering a smooth onboarding process can help encourage adoption.

Best regards,
Jaber

**Reply by jayceeday** [reply]
We are looking for human answers here.

**Comment by rainbow1016** [comment]
very good question!

**Comment by mscholz5** [comment]
I am an employee.
I wanted to use a trial version and share my experiences with my colleagues and superiors.

Your requirements for such a trial is conflicting with company policies.
I cannot decide about a payment plan or provide payment data right from the start.

So I gave up.

**Reply by jayceeday** [reply]
Thank you for the feedback, this is super helpful! 

What are the company policies that are conflicting? Is the policy you have to provide payment data from the start (Do you mean without a trial)?

**Reply by mscholz5** [reply]
Yes. Usually we start with an evaluation within out team. If we are convinced then we document the expected benefit and request a licence. Other persons will evaluate our request and decide about budget and payment plans.
Breaking that process means to involve more people right from the start, probably with an additional overhead of of discussions.
This makes a successful evaluation less feasable, as we cannot bring our personal experience into that discussion.

**Comment by trinitronx** [comment]
Can't try it out within VSCodium / Code OSS due to [this issue][1].


[1]: https://github.com/orgs/community/discussions/35832?sort=old

**Comment by BastianZim** [comment]
It's not available yet for my editor: https://github.com/orgs/community/discussions/7223

**Comment by denisdasilvarocha** [comment]
I did used it for couple of days, but dropped it fairly fast because it still used the Codex model. I know its now using the GPT 3.5, but im waiting for GPT-4 tho, i am getting way better results via ChatGPT+ on the site itself. 

Hope its soon, since OpenAI released the Business option now.

**Comment by clarkbox** [comment]
i am waiting for better support in Jetbrains Webstorm. tried switching over to VS for a week, but never got comfortable in it.

**Comment by monkeychambers** [comment]
I haven't paid for a subscription because guys do not allow payment via Paypal Wallet :(

**Comment by runephilosof-abtion** [comment]
Data leak concerns, missing [.copilotignore support](https://github.com/orgs/community/discussions/10305).

**Comment by xxwu-96** [comment]
Subject: Request for Manual Approval of GitHub Copilot

Dear GitHub Support Team,

I hope this email finds you well. I am writing to you as a contributor to an open-source project on GitHub and I would like to request manual approval for using GitHub Copilot.

I have been actively involved in the development of the following repository: https://github.com/gcpaas/DataRoom for the past three months. The project has gained considerable attention with 65 stars. Additionally, we have set up a demo environment for the project at: http://gcpaas.gccloud.com/bigScreen/big-screen-list.

I recently tried to access GitHub Copilot but unfortunately, the system's automatic detection flagged me as ineligible. As a dedicated contributor to the open-source community, I believe that GitHub Copilot would greatly enhance my productivity and the quality of my code contributions.

I would like to request your kind assistance in manually reviewing my eligibility for GitHub Copilot. I assure you that I will adhere to all the terms and conditions set forth by GitHub for using this tool.

Furthermore, I would like to mention that I have already applied for and have been using JetBrains All Products for the duration of one year. This demonstrates my commitment to utilizing the best development tools available to enhance my contributions.

I would greatly appreciate it if you could manually approve my access to GitHub Copilot. If there are any additional steps or iWechatIMG49.jpg
nformation required from my end, please do not hesitate to let me know.

Thank you for your attention and I look forward to your positive response.

Best regards,

**Reply by samus-aran** [reply]
Hi @xxwu-96 

We do not provide manual evaluations on the [eligibility for GitHub Copilot](https://docs.github.com/en/copilot/overview-of-github-copilot/about-github-copilot-for-individuals#about-billing-for-github-copilot). Feel free to use Copilot 30 day free trial. 👍

**Comment by cngarrison** [comment]
It's not available for my editor ([BBEdit](https://www.barebones.com/products/bbedit/)). 

BBEdit has excellent support for LSP. If you had a supported and well-documented LSP server then I expect I could use Copilot with my editor.

**Comment by mwaddell** [comment]
1. Security Concerns (CoPilot is not covered under any of GitHub's audits or certifications)
2. Copyright Concerns (CoPilot is trained on open-source software and we do not want GPLed code "contaminating" our proprietary technologies.  I haven't seen any indication that CoPilot was trained exclusively on code that is commercial-license-friendly (MIT, Apache, etc))
3. Transparency (It is unclear which code was contributed by CoPilot)
    - One idea for improving transparency I had is in #69717

Note: I have tried CoPilot for open-source work, but not for professional work because of the above concerns.

**Comment by PhilPoccia** [comment]
I want to be able to use it with my favorite text editor (which is BBEdit). They are restricted from incorporating GitHub Copilot into their application.

**Reply by Catchmaker** [reply]
I've pinged the BareBones team about AI coding tools a few times since I started using ChatGPT. For many of us, BBEdit is so productive that it's not worth switching to another editor.

**Reply by cngarrison** [reply]
This is my position as well. I have found AI-assisted coding to save me **heaps** of time. But using BBEdit saves me even **more** time. I won't switch editors for the sake of integrated AI. But I **will** choose an AI solution simply because of its well-supported editor integration.

**Comment by seiz** [comment]
I am also waiting for Support in BBedit, same as https://github.com/orgs/community/discussions/64222#discussioncomment-7217727

**Comment by Eiim** [comment]
1. Copyright concerns. I don't think I would use such an AI tool for writing public or freely-licensed code without a guarantee that all the training data is (eg) GPL-v3 compatible.
2. Editor support. Most of my code is in Eclipse or RStudio. Apparently there's an unofficial Eclipse plugin, but I hadn't heard of that until I just looked it up now. I doubt there'd be anything for RStudio, and I'd be surprised if it works well for R anyways, for multiple reasons. For the little bit of code that I use VSCodium for, it's not worth the effort.

**Comment by FJBDev** [comment]
Because there is no Eclipse Co-pilot plugin, as simple as that.

**Comment by xjjon** [comment]
We need Jetbrains Rider support

**Comment by SumitKumargiri** [comment]
> We are looking for human answers here.
😊😊

**Comment by norchai** [comment]
Also awaiting BBEdit support here.  Just spent a week trying out alternative editors specifically because I wanted to give Copilot a shot, but can't bring myself to give up some of BBEdit's ergonomics or unique features.

**Comment by jerabaul29** [comment]
I may be a bit late to the party, but this looks like a nice place to discuss :) .

I use the helix editor, and it is not so obvious how I should / could use copilot in it.

- if copilot could be available as a general "standardized" LSP, I would just add it to my config; but so far I could not find a github supported "official copilot LSP" that is compliant with the LSP standard (but I may be missing something?)

- it also looks like copilot may require more than "just" a LSP interface to reach its highest potential - maybe a "AI IDE protocol", a bit akin to the LSP protocol but thought for use with AI assistants, with additional functionality thought for this case, would be nice? See for example https://github.com/helix-editor/helix/discussions/4037 .

At the moment, I am looking at different options to use copilot in helix editor - for example, investigating the use of things like https://github.com/leona/helix-gpt , or https://github.com/TerminalFi/LSP-copilot . But something officially supported by Github would be welcome! :)

**Comment by elthariel** [comment]
Because it's a Microsoft product and I'm trying to limit my exposure to this risk
